met-web/src/components/engagement/form/EngagementWidgets/Subscribe/FormSignUpDrawer.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useContext, useState, useEffect } from 'react';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import Divider from '@mui/material/Divider';
+import { Grid, FormGroup, FormControlLabel, Radio } from '@mui/material';
+import { MetHeader3, MetLabel, PrimaryButton, SecondaryButton } from 'components/common';
+import { useForm, FormProvider, SubmitHandler, Controller, ControllerRenderProps } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useAppDispatch } from 'hooks';
+import { SubscribeContext } from './SubscribeContext';
+import ControlledTextField from 'components/common/ControlledInputComponents/ControlledTextField';
+import { Subscribe_TYPE } from 'models/subscription';
+import RichTextEditor from 'components/common/RichTextEditor';
+
+const schema = yup
+    .object({
+        description: yup.string().max(500, 'Description cannot exceed 500 characters'),
+        cta_type: yup.string(),
+        cta_text: yup.string().max(25, 'Description cannot exceed 25 characters'),
+    })
+    .required();
+
+type FormSignUp = yup.TypeOf<typeof schema> & {
+    cta_type: 'link' | 'button';
+};
+
+const FormSignUpDrawer = () => {
+    const dispatch = useAppDispatch();
+    const { handleSubscribeDrawerOpen, formSignUpTabOpen, richFormSignUpDescription, setRichFormSignUpDescription } =
+        useContext(SubscribeContext);
+    const [isCreating, setIsCreating] = useState(false);
+    const [initialRichDescription, setInitialRichDescription] = useState('');
+    const [descriptionCharCount, setDescriptionCharCount] = useState(0);
+    const methods = useForm<FormSignUp>({
+        resolver: yupResolver(schema),
+    });
+
+    const getTextFromDraftJsContentState = (contentJSON: string): string => {
```

Comment:
Theres's a utils file in the rich text editor folder, could you check if this exists there already and if not to add it there

Reply:
it does not but I added it there as you suggested